pkg/cluster/snow.go

```diff
@@ -268,3 +271,21 @@ func ValidateSnowMachineRefExists(c *Config) error {
 	}
 	return nil
 }
+
+func validateSnowUnstackedEtcd(c *Config) error {
+	if c.Cluster.Spec.ExternalEtcdConfiguration == nil || c.Cluster.Spec.ExternalEtcdConfiguration.MachineGroupRef == nil {
+		return nil
+	}
+
+	mc := c.SnowMachineConfig(c.Cluster.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name)
```

Comment:
Why can this be nil?

Reply:
on testing mainly. also bc mc is a pointer, it can be nil. i don't wanna this method to depend on side affect